packages/mui-material/src/PaginationItem/PaginationItem.js

```diff
@@ -151,115 +158,142 @@ const PaginationItemPage = styled(ButtonBase, {
+                  : alpha(
+                      theme.palette[color].main,
+                      theme.palette.action.activatedOpacity + theme.palette.action.focusOpacity,
+                    ),
+                // Reset on touch devices, it doesn't add specificity
+                '@media (hover: none)': {
+                  backgroundColor: 'transparent',
+                },
+              },
+              [`&.${paginationItemClasses.focusVisible}`]: {
+                backgroundColor: theme.vars
+                  ? `rgba(${theme.vars.palette[color].mainChannel} / calc(${theme.vars.palette.action.activatedOpacity} + ${theme.vars.palette.action.focusOpacity}))`
+                  : alpha(
+                      theme.palette[color].main,
+                      theme.palette.action.activatedOpacity + theme.palette.action.focusOpacity,
+                    ),
+              },
+              [`&.${paginationItemClasses.disabled}`]: {
+                borderColor: (theme.vars || theme).palette.action.disabledBackground,
+                color: (theme.vars || theme).palette.action.disabled,
+              },
             },
           },
-          [`&.${paginationItemClasses.focusVisible}`]: {
-            backgroundColor: theme.vars
-              ? `rgba(${theme.vars.palette[ownerState.color].mainChannel} / calc(${
-                  theme.vars.palette.action.activatedOpacity
-                } + ${theme.vars.palette.action.focusOpacity}))`
-              : alpha(
-                  theme.palette[ownerState.color].main,
-                  theme.palette.action.activatedOpacity + theme.palette.action.focusOpacity,
-                ),
-          },
-        }),
-        [`&.${paginationItemClasses.disabled}`]: {
-          borderColor: (theme.vars || theme).palette.action.disabledBackground,
-          color: (theme.vars || theme).palette.action.disabled,
-        },
-      },
-    }),
+        })),
+    ],
   }),
 );
 
 const PaginationItemPageIcon = styled('div', {
   name: 'MuiPaginationItem',
   slot: 'Icon',
-  overridesResolver: (props, styles) => styles.icon,
-})(({ theme, ownerState }) => ({
+  overridesResolver: (_props, styles) => styles.icon,
+})(({ theme }) => ({
   fontSize: theme.typography.pxToRem(20),
   margin: '0 -8px',
-  ...(ownerState.size === 'small' && {
-    fontSize: theme.typography.pxToRem(18),
-  }),
-  ...(ownerState.size === 'large' && {
-    fontSize: theme.typography.pxToRem(22),
-  }),
+  variants: [
+    {
+      props: { size: 'small' },
+      style: {
+        fontSize: theme.typography.pxToRem(18),
+      },
+    },
+    {
+      props: { size: 'large' },
+      style: {
+        fontSize: theme.typography.pxToRem(22),
+      },
+    },
+  ],
 }));
 
 const PaginationItem = React.forwardRef(function PaginationItem(inProps, ref) {
+  const useThemeProps = createUseThemeProps('MuiPaginationItem');
```

Comment:
Please move out of the component to after imports.

Reply:
I'll do the same for my other PR's 👍